src/reactionsystem.jl

```diff
@@ -366,6 +369,45 @@ function raterule_to_diffeq(model, rule)
     end
 end
 
+function constant_to_diffeq(species_id)
+    D = Differential(Catalyst.DEFAULT_IV)
+    var = create_var(species_id, Catalyst.DEFAULT_IV)
+    return D(var) ~ 0
+end
+
+function get_underdetermined_species(model)
+    rules = model.rules
+    defined_species = String[r.id for r in rules if r isa Union{SBML.AssignmentRule, SBML.RateRule}]
+    species_changed_by_reactions = get_species_changed_by_reaction(model)
+    undefined_species = [k for (k,v) in model.species if !(k in vcat(defined_species, species_changed_by_reactions)) && v.constant!=true]
+    Equation[constant_to_diffeq(s) for s in undefined_species]      
+end
+
+function get_species_changed_by_reaction(model)
+    species_changed_by_reaction = Set([])
+    for reaction in values(model.reactions)
+        rstoich = reaction.reactants
+        pstoich = reaction.products
+        netstoich = Dict()
+        for (k, v) in rstoich
+            netstoich[k] = v
+        end
+        for (k, v) in pstoich
+            if haskey(netstoich, k) && netstoich[k] == v
+                delete!(netstoich, k)
+            else
```

Comment:
Is the value in `netstoich` even used subsequent? If not you could drop the `else` clause.

Otherwise all looks good to me.

Reply:
No. I could do sth like:
```julia
function get_species_changed_by_reaction(model)
    species_changed_by_reaction = Set([])
    for reaction in values(model.reactions)
        rstoich = reaction.reactants
        pstoich = reaction.products
        candidates = collect(keys(rstoich))
        for (k, v) in pstoich
            haskey(rstoich, k) && rstoich[k] == v ? delete!(candidates, k) : push!(candidates, k)
        end
        for specie in candidates
            if !model.species[specie].boundary_condition && !model.species[specie].constant
                push!(species_changed_by_reaction, specie)
            end
        end
    end
    collect(species_changed_by_reaction)
end
```

Reply:
I've actually done that just now.